### ISSUE TITLE
[JENKINS-46207] git-plugin should support tags in multibranch pipeline configurations

### ISSUE THREAD

**Issue by jenkins-infra-bot** [jenkinsci/git-plugin#3175]
<p>Currently the git plugin doesn't seem to support tags in multi-branch builds.  It looks like this is because the ref prefixes are hard-coded. See: <a href="https://github.com/jenkinsci/git-plugin/blob/master/src/main/java/jenkins/plugins/git/AbstractGitSCMSource.java#L417" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkinsci/git-plugin/blob/master/src/main/java/jenkins/plugins/git/AbstractGitSCMSource.java#L417</a></p>

<p>This is similar to <a href="https://issues.jenkins.io/browse/JENKINS-34395" title="Support for building tags" class="issue-link" data-issue-key="JENKINS-34395"><del>JENKINS-34395</del></a> but for the git plugin.  I'm creating a  separate issue because the causes and the fixes appear to be unrelated.</p>

---
<details><summary><i>Originally reported by <img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/kbrowder.jpg" title="kbrowder's avatar" /> <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=kbrowder">kbrowder</a>, imported from: <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-46207" target="_blank">git-plugin should support tags in multibranch pipeline configurations</a></i></summary>
<i><ul>
<li><b>assignee</b>: <img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/stephenconnolly.png" title="stephenconnolly's avatar" /> <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=stephenconnolly">stephenconnolly</a>
<li><b>status</b>: Closed
<li><b>priority</b>: Minor
<li><b>component(s)</b>: git-plugin
<li><b>resolution</b>: Fixed
<li><b>resolved</b>: 2017-10-16T14:22:35+00:00
<li><b>votes</b>: 3
<li><b>watchers</b>: 10
<li><b>imported</b>: 2025-12-02
</ul></i>
<details><summary>Raw content of original issue</summary>

<pre>
<p>Currently the git plugin doesn't seem to support tags in multi-branch builds.  It looks like this is because the ref prefixes are hard-coded. See: <a href="https://github.com/jenkinsci/git-plugin/blob/master/src/main/java/jenkins/plugins/git/AbstractGitSCMSource.java#L417" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkinsci/git-plugin/blob/master/src/main/java/jenkins/plugins/git/AbstractGitSCMSource.java#L417</a></p>

<p>This is similar to <a href="https://issues.jenkins.io/browse/JENKINS-34395" title="Support for building tags" class="issue-link" data-issue-key="JENKINS-34395"><del>JENKINS-34395</del></a> but for the git plugin.  I'm creating a  separate issue because the causes and the fixes appear to be unrelated.</p></pre>
</details>
</details>

**Comment by jenkins-infra-bot** [comment]
- _Remote link associated with this issue:_


  - [PR#​528](https://github.com/jenkinsci/git-plugin/pull/528)

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/kbrowder.jpg" title="kbrowder's avatar" /> kbrowder:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-46207?focusedId=310495&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-310495">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Oh that's weird JIRA said it lost this when i created it, apparently it didn't, marking as duplicate.  My Applologies</p>
</pre>
</details>

</ul>
</details>

<p>Oh that's weird JIRA said it lost this when i created it, apparently it didn't, marking as duplicate.  My Applologies</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/stephenconnolly.png" title="stephenconnolly's avatar" /> stephenconnolly:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-46207?focusedId=312349&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-312349">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Prefer this one to <a href="https://issues.jenkins.io/browse/JENKINS-46208" title="git-plugin doesn't support tags" class="issue-link" data-issue-key="JENKINS-46208"><del>JENKINS-46208</del></a> as this has a better description</p>
</pre>
</details>

</ul>
</details>

<p>Prefer this one to <a href="https://issues.jenkins.io/browse/JENKINS-46208" title="git-plugin doesn't support tags" class="issue-link" data-issue-key="JENKINS-46208"><del>JENKINS-46208</del></a> as this has a better description</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/stephenconnolly.png" title="stephenconnolly's avatar" /> stephenconnolly:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-46207?focusedId=312350&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-312350">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Nothing to do with ref-specs, the actual impl is more complex.</p>

<p>My initial stab at an implementation is <a href="https://github.com/jenkinsci/git-plugin/pull/528" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkinsci/git-plugin/pull/528</a></p>

<p>Will need testing before we can merge... will likely need changes in branch-api to prevent build storms unless I was good at anticipating the needs of tags</p>
</pre>
</details>

</ul>
</details>

<p>Nothing to do with ref-specs, the actual impl is more complex.</p>

<p>My initial stab at an implementation is <a href="https://github.com/jenkinsci/git-plugin/pull/528" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkinsci/git-plugin/pull/528</a></p>

<p>Will need testing before we can merge... will likely need changes in branch-api to prevent build storms unless I was good at anticipating the needs of tags</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/stephenconnolly.png" title="stephenconnolly's avatar" /> stephenconnolly:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-46207?focusedId=312351&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-312351">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p><a href="https://github.com/jenkinsci/branch-api-plugin/commit/cf848af4addf33f0aae9c86149fc0cfd1fee5c82#diff-b347159a8a8075d4e700065f8668b804R1811" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkinsci/branch-api-plugin/commit/cf848af4addf33f0aae9c86149fc0cfd1fee5c82#diff-b347159a8a8075d4e700065f8668b804R1811</a> Looks like a was good at anticipating the needs of tags... ok just requires testing!</p>
</pre>
</details>

</ul>
</details>

<p><a href="https://github.com/jenkinsci/branch-api-plugin/commit/cf848af4addf33f0aae9c86149fc0cfd1fee5c82#diff-b347159a8a8075d4e700065f8668b804R1811" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkinsci/branch-api-plugin/commit/cf848af4addf33f0aae9c86149fc0cfd1fee5c82#diff-b347159a8a8075d4e700065f8668b804R1811</a> Looks like a was good at anticipating the needs of tags... ok just requires testing!</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/stephenconnolly.png" title="stephenconnolly's avatar" /> stephenconnolly:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-46207?focusedId=312353&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-312353">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Manual testing looks good!</p>
</pre>
</details>

</ul>
</details>

<p>Manual testing looks good!</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/kbrowder.jpg" title="kbrowder's avatar" /> kbrowder:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-46207?focusedId=313918&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-313918">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Very cool, simpler than what I was thinking too, thanks <img class="emoticon" src="https://issues.jenkins.io/images/icons/emoticons/smile.png" height="16" width="16" align="absmiddle" alt="" border="0"/></p>
</pre>
</details>

</ul>
</details>

<p>Very cool, simpler than what I was thinking too, thanks <img class="emoticon" src="https://issues.jenkins.io/images/icons/emoticons/smile.png" height="16" width="16" align="absmiddle" alt="" border="0"/></p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/scm_issue_link.png" title="scm_issue_link's avatar" /> scm_issue_link:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-46207?focusedId=315521&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-315521">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Code changed in jenkins
User: Stephen Connolly
Path:
 src/main/java/jenkins/plugins/git/AbstractGitSCMSource.java
 src/main/java/jenkins/plugins/git/GitSCMSource.java
 src/main/java/jenkins/plugins/git/GitTagSCMHead.java
 src/main/java/jenkins/plugins/git/GitTagSCMRevision.java
 src/main/java/jenkins/plugins/git/traits/BranchDiscoveryTrait.java
 src/main/java/jenkins/plugins/git/traits/TagDiscoveryTrait.java
 src/main/resources/jenkins/plugins/git/traits/Messages.properties
 src/main/resources/jenkins/plugins/git/traits/TagDiscoveryTrait/config.jelly
 src/main/resources/jenkins/plugins/git/traits/TagDiscoveryTrait/help.html
 src/test/java/jenkins/plugins/git/AbstractGitSCMSourceTest.java
<a href="http://jenkins-ci.org/commit/git-plugin/837ae6424c7636b6f7aaabbe8337ae3a2c053d0a" class="external-link" target="_blank" rel="nofollow noopener">http://jenkins-ci.org/commit/git-plugin/837ae6424c7636b6f7aaabbe8337ae3a2c053d0a</a>
Log:
  <a href="https://issues.jenkins.io/browse/JENKINS-46207" title="git-plugin should support tags in multibranch pipeline configurations" class="issue-link" data-issue-key="JENKINS-46207"><del>JENKINS-46207</del></a> Initial implementation, if lucky this is complete</p>

<ul class="alternate" type="square">
	<li>I suspect there are additional changes required before can merge for git</li>
	<li>I suspect there are additional changes required to branch-api before safe to merge for users</li>
</ul>



</pre>
</details>

</ul>
</details>

<p>Code changed in jenkins<br/>
User: Stephen Connolly<br/>
Path:<br/>
 src/main/java/jenkins/plugins/git/AbstractGitSCMSource.java<br/>
 src/main/java/jenkins/plugins/git/GitSCMSource.java<br/>
 src/main/java/jenkins/plugins/git/GitTagSCMHead.java<br/>
 src/main/java/jenkins/plugins/git/GitTagSCMRevision.java<br/>
 src/main/java/jenkins/plugins/git/traits/BranchDiscoveryTrait.java<br/>
 src/main/java/jenkins/plugins/git/traits/TagDiscoveryTrait.java<br/>
 src/main/resources/jenkins/plugins/git/traits/Messages.properties<br/>
 src/main/resources/jenkins/plugins/git/traits/TagDiscoveryTrait/config.jelly<br/>
 src/main/resources/jenkins/plugins/git/traits/TagDiscoveryTrait/help.html<br/>
 src/test/java/jenkins/plugins/git/AbstractGitSCMSourceTest.java<br/>
<a href="http://jenkins-ci.org/commit/git-plugin/837ae6424c7636b6f7aaabbe8337ae3a2c053d0a" class="external-link" target="_blank" rel="nofollow noopener">http://jenkins-ci.org/commit/git-plugin/837ae6424c7636b6f7aaabbe8337ae3a2c053d0a</a><br/>
Log:<br/>
  <a href="https://issues.jenkins.io/browse/JENKINS-46207" title="git-plugin should support tags in multibranch pipeline configurations" class="issue-link" data-issue-key="JENKINS-46207"><del>JENKINS-46207</del></a> Initial implementation, if lucky this is complete</p>

<ul class="alternate" type="square">
	<li>I suspect there are additional changes required before can merge for git</li>
	<li>I suspect there are additional changes required to branch-api before safe to merge for users</li>
</ul>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/scm_issue_link.png" title="scm_issue_link's avatar" /> scm_issue_link:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-46207?focusedId=315522&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-315522">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Code changed in jenkins
User: Mark Waite
Path:
 src/main/java/jenkins/plugins/git/AbstractGitSCMSource.java
 src/main/java/jenkins/plugins/git/GitSCMSource.java
 src/main/java/jenkins/plugins/git/GitTagSCMHead.java
 src/main/java/jenkins/plugins/git/GitTagSCMRevision.java
 src/main/java/jenkins/plugins/git/traits/BranchDiscoveryTrait.java
 src/main/java/jenkins/plugins/git/traits/TagDiscoveryTrait.java
 src/main/resources/jenkins/plugins/git/traits/Messages.properties
 src/main/resources/jenkins/plugins/git/traits/TagDiscoveryTrait/config.jelly
 src/main/resources/jenkins/plugins/git/traits/TagDiscoveryTrait/help.html
 src/test/java/jenkins/plugins/git/AbstractGitSCMSourceTest.java
<a href="http://jenkins-ci.org/commit/git-plugin/fe5a3f42a1f7e6ceb978ee2e603a1068dfeba970" class="external-link" target="_blank" rel="nofollow noopener">http://jenkins-ci.org/commit/git-plugin/fe5a3f42a1f7e6ceb978ee2e603a1068dfeba970</a>
Log:
  Merge pull request #528 from stephenc/jenkins-46207</p>

<p><a href="https://issues.jenkins.io/browse/JENKINS-46207" title="git-plugin should support tags in multibranch pipeline configurations" class="issue-link" data-issue-key="JENKINS-46207"><del>JENKINS-46207</del></a> Enable tag discovery</p>


<p>Compare: <a href="https://github.com/jenkinsci/git-plugin/compare/35745f2e3f81...fe5a3f42a1f7" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkinsci/git-plugin/compare/35745f2e3f81...fe5a3f42a1f7</a></p>

</pre>
</details>

</ul>
</details>

<p>Code changed in jenkins<br/>
User: Mark Waite<br/>
Path:<br/>
 src/main/java/jenkins/plugins/git/AbstractGitSCMSource.java<br/>
 src/main/java/jenkins/plugins/git/GitSCMSource.java<br/>
 src/main/java/jenkins/plugins/git/GitTagSCMHead.java<br/>
 src/main/java/jenkins/plugins/git/GitTagSCMRevision.java<br/>
 src/main/java/jenkins/plugins/git/traits/BranchDiscoveryTrait.java<br/>
 src/main/java/jenkins/plugins/git/traits/TagDiscoveryTrait.java<br/>
 src/main/resources/jenkins/plugins/git/traits/Messages.properties<br/>
 src/main/resources/jenkins/plugins/git/traits/TagDiscoveryTrait/config.jelly<br/>
 src/main/resources/jenkins/plugins/git/traits/TagDiscoveryTrait/help.html<br/>
 src/test/java/jenkins/plugins/git/AbstractGitSCMSourceTest.java<br/>
<a href="http://jenkins-ci.org/commit/git-plugin/fe5a3f42a1f7e6ceb978ee2e603a1068dfeba970" class="external-link" target="_blank" rel="nofollow noopener">http://jenkins-ci.org/commit/git-plugin/fe5a3f42a1f7e6ceb978ee2e603a1068dfeba970</a><br/>
Log:<br/>
  Merge pull request #528 from stephenc/jenkins-46207</p>

<p><a href="https://issues.jenkins.io/browse/JENKINS-46207" title="git-plugin should support tags in multibranch pipeline configurations" class="issue-link" data-issue-key="JENKINS-46207"><del>JENKINS-46207</del></a> Enable tag discovery</p>


<p>Compare: <a href="https://github.com/jenkinsci/git-plugin/compare/35745f2e3f81...fe5a3f42a1f7" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkinsci/git-plugin/compare/35745f2e3f81...fe5a3f42a1f7</a></p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/markewaite.png" title="markewaite's avatar" /> markewaite:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-46207?focusedId=316983&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-316983">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Released in <a href="https://wiki.jenkins.io/display/JENKINS/Git+Plugin#GitPlugin-ChangeLog" class="external-link" target="_blank" rel="nofollow noopener">git plugin 3.6.0</a></p>
</pre>
</details>

</ul>
</details>

<p>Released in <a href="https://wiki.jenkins.io/display/JENKINS/Git+Plugin#GitPlugin-ChangeLog" class="external-link" target="_blank" rel="nofollow noopener">git plugin 3.6.0</a></p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>kr:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-46207?focusedId=317069&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-317069">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>the solution does not work for me, I commented the problem to the PR: <a href="https://github.com/jenkinsci/git-plugin/pull/528" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkinsci/git-plugin/pull/528</a></p>
</pre>
</details>

</ul>
</details>

<p>the solution does not work for me, I commented the problem to the PR: <a href="https://github.com/jenkinsci/git-plugin/pull/528" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkinsci/git-plugin/pull/528</a></p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>kr:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-46207?focusedId=317876&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-317876">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>with git-plugin 3.6.2 tags are successfully discovered, however they are not automatically built.</p>

<p>Output of the scan multibranch pipeline log:  </p>
<div class="code panel" style="border-width: 1px;"><div class="codeContent panelContent">
<pre class="code-java">Checking tag v0.10.0
<span class="code-quote">'Jenkinsfile'</span> found
Met criteria
No automatic builds <span class="code-keyword">for</span> v0.10.0
</pre>
</div></div>
<p>the reason is that Multibranch Pipeline uses heads only to schedule automatic builds:
<a href="https://github.com/jenkinsci/branch-api-plugin/blob/master/src/main/java/jenkins/branch/MultiBranchProject.java#L2151" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkinsci/branch-api-plugin/blob/master/src/main/java/jenkins/branch/MultiBranchProject.java#L2151</a></p>

<p>is it intended behaviour?</p>
</pre>
</details>

</ul>
</details>

<p>with git-plugin 3.6.2 tags are successfully discovered, however they are not automatically built.</p>

<p>Output of the scan multibranch pipeline log:  </p>

<pre>
Checking tag v0.10.0
<span class="code-quote">'Jenkinsfile'</span> found
Met criteria
No automatic builds <span class="code-keyword">for</span> v0.10.0
</pre>
<p>the reason is that Multibranch Pipeline uses heads only to schedule automatic builds:<br/>
<a href="https://github.com/jenkinsci/branch-api-plugin/blob/master/src/main/java/jenkins/branch/MultiBranchProject.java#L2151" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkinsci/branch-api-plugin/blob/master/src/main/java/jenkins/branch/MultiBranchProject.java#L2151</a></p>

<p>is it intended behaviour?</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/stephenconnolly.png" title="stephenconnolly's avatar" /> stephenconnolly:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-46207?focusedId=317890&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-317890">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=kr" class="user-hover" rel="kr">kr</a></p>

<p>It is intended behaviour... look a few lines further down: <a href="https://github.com/jenkinsci/branch-api-plugin/blob/master/src/main/java/jenkins/branch/MultiBranchProject.java#L2165" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkinsci/branch-api-plugin/blob/master/src/main/java/jenkins/branch/MultiBranchProject.java#L2165</a></p>
<ul>
	<li>Some people may configure the Jenkinsfile to deploy to production for tags.</li>
	<li>We cannot control the order in which tags will be built if multiple tags are discovered in the same indexing</li>
	<li>As a result, building tags automatically would be risky.</li>
</ul>


<p>There is an extension point (which is hidden until you have some implementations) called <tt>BranchBuildStrategy</tt> and that can determine if you want to build tags automatically, but this would be <b>opt-in</b> behaviour.</p>

<p>By default, tags will not be built. You must configure automatic building of tags... see <a href="https://github.com/jenkinsci/github-branch-source-plugin/pull/158#issuecomment-332773194" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkinsci/github-branch-source-plugin/pull/158#issuecomment-332773194</a> and <a href="https://github.com/AngryBytes/jenkins-build-everything-strategy-plugin" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/AngryBytes/jenkins-build-everything-strategy-plugin</a> for starting points if you want to build tags automatically</p>
</pre>
</details>

</ul>
</details>

<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=kr" class="user-hover" rel="kr">kr</a></p>

<p>It is intended behaviour... look a few lines further down: <a href="https://github.com/jenkinsci/branch-api-plugin/blob/master/src/main/java/jenkins/branch/MultiBranchProject.java#L2165" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkinsci/branch-api-plugin/blob/master/src/main/java/jenkins/branch/MultiBranchProject.java#L2165</a></p>
<ul>
	<li>Some people may configure the Jenkinsfile to deploy to production for tags.</li>
	<li>We cannot control the order in which tags will be built if multiple tags are discovered in the same indexing</li>
	<li>As a result, building tags automatically would be risky.</li>
</ul>


<p>There is an extension point (which is hidden until you have some implementations) called <tt>BranchBuildStrategy</tt> and that can determine if you want to build tags automatically, but this would be <b>opt-in</b> behaviour.</p>

<p>By default, tags will not be built. You must configure automatic building of tags... see <a href="https://github.com/jenkinsci/github-branch-source-plugin/pull/158#issuecomment-332773194" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkinsci/github-branch-source-plugin/pull/158#issuecomment-332773194</a> and <a href="https://github.com/AngryBytes/jenkins-build-everything-strategy-plugin" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/AngryBytes/jenkins-build-everything-strategy-plugin</a> for starting points if you want to build tags automatically</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>hamiltb:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-46207?focusedId=320016&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-320016">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=stephenconnolly" class="user-hover" rel="stephenconnolly">stephenconnolly</a> <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=markwaite" class="user-hover" rel="markwaite">markwaite</a> If I kick off a tag build, do you know how in the Jenkinsfile I can detect that it is a tag build?</p>
</pre>
</details>

</ul>
</details>

<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=stephenconnolly" class="user-hover" rel="stephenconnolly">stephenconnolly</a> <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=markwaite" class="user-hover" rel="markwaite">markwaite</a> If I kick off a tag build, do you know how in the Jenkinsfile I can detect that it is a tag build?</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/stephenconnolly.png" title="stephenconnolly's avatar" /> stephenconnolly:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-46207?focusedId=320018&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-320018">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>The <tt>env.TAG_NAME</tt> will be non-null when building a tag</p>
</pre>
</details>

</ul>
</details>

<p>The <tt>env.TAG_NAME</tt> will be non-null when building a tag</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/markewaite.png" title="markewaite's avatar" /> markewaite:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-46207?focusedId=320029&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-320029">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=hamiltb" class="user-hover" rel="hamiltb">hamiltb</a> if you need to determine the list of all tags which point at a specific SHA1, you can use the shell command:</p>

<p><tt>git tag --points-at HEAD</tt></p>

<p>The <tt>env.TAG_NAME</tt> which <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=stephenconnolly" class="user-hover" rel="stephenconnolly">stephenconnolly</a> references should be more reliable than the <tt>points-at</tt> shell trick, since it is set inside the plugins, rather than being guessed from the condition of the repository.</p>
</pre>
</details>

</ul>
</details>

<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=hamiltb" class="user-hover" rel="hamiltb">hamiltb</a> if you need to determine the list of all tags which point at a specific SHA1, you can use the shell command:</p>

<p><tt>git tag --points-at HEAD</tt></p>

<p>The <tt>env.TAG_NAME</tt> which <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=stephenconnolly" class="user-hover" rel="stephenconnolly">stephenconnolly</a> references should be more reliable than the <tt>points-at</tt> shell trick, since it is set inside the plugins, rather than being guessed from the condition of the repository.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>hamiltb:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-46207?focusedId=320087&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-320087">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=stephenconnolly" class="user-hover" rel="stephenconnolly">stephenconnolly</a> <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=markewaite" class="user-hover" rel="markewaite">markewaite</a></p>

<p>env.TAG_NAME doesnt exist in a tag build. I printed out all envars:</p>

<p><font color="#59afe1"><em>@NonCPS</em></font>
 <font color="#59afe1"><em>def printParams() {</em></font>
 <font color="#59afe1"> <em>env.getEnvironment().each { name, value -&gt; println "$name : $value" }</em></font>
 <font color="#59afe1"><em>}</em></font>
 <font color="#59afe1"><em>printParams()</em></font></p>

<p>and TAG_NAME isnt there.</p>
</pre>
</details>

</ul>
</details>

<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=stephenconnolly" class="user-hover" rel="stephenconnolly">stephenconnolly</a> <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=markewaite" class="user-hover" rel="markewaite">markewaite</a></p>

<p>env.TAG_NAME doesnt exist in a tag build. I printed out all envars:</p>

<p><font color="#59afe1"><em>@​NonCPS</em></font><br/>
 <font color="#59afe1"><em>def printParams() {</em></font><br/>
 <font color="#59afe1"> <em>env.getEnvironment().each { name, value -> println "$name : $value" }</em></font><br/>
 <font color="#59afe1"><em>}</em></font><br/>
 <font color="#59afe1"><em>printParams()</em></font></p>

<p>and TAG_NAME isnt there.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/stephenconnolly.png" title="stephenconnolly's avatar" /> stephenconnolly:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-46207?focusedId=320122&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-320122">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>What version of Branch API? TAG_NAME was added in <a href="https://wiki.jenkins.io/display/JENKINS/Branch+API+Plugin#BranchAPIPlugin-Version2.0.11(Jul17,2017)" class="external-link" target="_blank" rel="nofollow noopener">2.0.11</a></p>
</pre>
</details>

</ul>
</details>

<p>What version of Branch API? TAG_NAME was added in <a href="https://wiki.jenkins.io/display/JENKINS/Branch+API+Plugin#BranchAPIPlugin-Version2.0.11(Jul17,2017)" class="external-link" target="_blank" rel="nofollow noopener">2.0.11</a></p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>hamiltb:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-46207?focusedId=320309&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-320309">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Yep, works good now, thanks!</p>
</pre>
</details>

</ul>
</details>

<p>Yep, works good now, thanks!</p>

**Comment by jenkins-infra-bot** [comment]
<i>[Original `is_duplicated_by` from Jira: <a href="https://github.com/jenkinsci/git-plugin/issues?q=is%3Aissue%20%22jira_issue_key%3DJENKINS-46208%22">JENKINS-46208</a>]</i>

**Comment by jenkins-infra-bot** [comment]
<i>[Original `is_duplicated_by` from Jira: <a href="https://github.com/jenkinsci/git-plugin/issues?q=is%3Aissue%20%22jira_issue_key%3DJENKINS-37050%22">JENKINS-37050</a>]</i>

**Comment by jenkins-infra-bot** [comment]
<i>[Original `relates_to` from Jira: <a href="https://github.com/jenkinsci/git-plugin/issues?q=is%3Aissue%20%22jira_issue_key%3DJENKINS-34395%22">JENKINS-34395</a>]</i>

**Comment by jenkins-infra-bot** [comment]
<i>[Original `relates_to` from Jira: <a href="https://github.com/jenkinsci/git-plugin/issues?q=is%3Aissue%20%22jira_issue_key%3DJENKINS-47254%22">JENKINS-47254</a>]</i>
